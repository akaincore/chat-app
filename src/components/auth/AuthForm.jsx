import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    marginTop: '20px',
    marginBottom: '20px',
    width: '80%',
  },
  authForm: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  submitButton: {
    margin: theme.spacing.unit,
    width: 200,
    marginTop: '20px',
    marginBottom: '20px',
  },
  hidden: {
    display: 'none',
  },
});

const AuthForm = ({
 fields, values, submitText, classes, inputChange, onSubmit,
}) => (
  <form className={classes.authForm} onSubmit={onSubmit}>
    {fields.map(({
 type, name, placeholder, required,
}) => (
      <TextField
        fullWidth
        key={`${type}-${name}`}
        label={placeholder}
        placeholder={placeholder}
        name={name}
        value={values[name]}
        type={type}
        required={required}
        className={classes.textField}
        margin="normal"
        onChange={inputChange}
      />
    ))}
    <Button variant="contained" color="primary" className={classes.submitButton} type="submit">
      {submitText}
    </Button>
  </form>
);

AuthForm.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  fields: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
      placeholder: PropTypes.string.isRequired,
      required: PropTypes.bool.isRequired,
    }),
  ).isRequired,
  values: PropTypes.objectOf(PropTypes.string).isRequired,
  submitText: PropTypes.string.isRequired,
  inputChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default withStyles(styles)(AuthForm);
