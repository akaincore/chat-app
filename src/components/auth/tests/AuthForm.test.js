/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AuthForm from '../AuthForm';

const mockProps = {
  fields: [
    {
      name: 'field',
      type: 'text',
      placeholder: 'placeholder',
      required: true,
    },
  ],
  values: { field: 'baz' },
  submitText: 'text',
  inputChange: jest.fn(),
  onSubmit: jest.fn(),
};

describe('<AuthForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AuthForm {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AuthForm {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
