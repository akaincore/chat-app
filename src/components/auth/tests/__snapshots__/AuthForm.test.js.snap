// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthForm /> renders correctly 1`] = `
<form
  className="AuthForm-authForm-94"
  onSubmit={[MockFunction]}
>
  <div
    className="MuiFormControl-root-97 MuiFormControl-marginNormal-98 MuiFormControl-fullWidth-100 AuthForm-textField-93"
  >
    <label
      className="MuiFormLabel-root-112 MuiFormLabel-filled-116 MuiFormLabel-required-117 MuiInputLabel-required-105 MuiInputLabel-root-101 MuiInputLabel-formControl-106 MuiInputLabel-animated-109 MuiInputLabel-shrink-108"
      data-shrink={true}
    >
      placeholder
      <span
        className="MuiFormLabel-asterisk-118"
      >
        â€‰*
      </span>
    </label>
    <div
      className="MuiInputBase-root-132 MuiInput-root-119 MuiInput-underline-123 MuiInputBase-fullWidth-141 MuiInput-fullWidth-126 MuiInputBase-formControl-133 MuiInput-formControl-120"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input-142 MuiInput-input-127"
        disabled={false}
        name="field"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="placeholder"
        required={true}
        type="text"
        value="baz"
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-175 MuiButton-root-149 MuiButton-contained-160 MuiButton-containedPrimary-161 MuiButton-raised-163 MuiButton-raisedPrimary-164 AuthForm-submitButton-95"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-150"
    >
      text
    </span>
    <span
      className="MuiTouchRipple-root-178"
    />
  </button>
</form>
`;
